ui/src/main/webapp/src/app/shared/checkboxes.component.ts

```diff
@@ -24,168 +22,72 @@ export class CheckboxesComponent implements OnInit, OnChanges
 
-    public constructor(element: ElementRef, private _zone: NgZone) {
+    public constructor(element: ElementRef) {
         this.component = this;
         this.rootElement = element.nativeElement;
     }
 
-    public ngOnChanges(changes: {[options: string]: SimpleChange}): any {
-        console.warn("onChanges", changes['options']);
-
-        /* /// This is handled by the setter?
-        if (changes.hasOwnProperty('options')) {
-            let newOptions: CheckboxData[] = changes['options'].currentValue;
-            this.options = newOptions;
-        }
-        */
-
-        if (changes.hasOwnProperty('checkedOptions')) {
-            this.checkedOptions = changes['checkedOptions'].currentValue;
-        }
-
-        ///this.setChangeHandlersToCheckboxes();
-    }
-
-    public ngOnInit() {
-        //this.setChangeHandersToCheckboxes();
-    }
-
-    private setChangeHandlersToCheckboxes() {
-        console.log("setChangeHandersToCheckboxes() called");
-        /*setTimeout(() => {
-            $(this.rootElement).find(":checkbox")
-                .off()
-                .change((event, data) => this.updateCheckedValues(event, data));
-        }, 2000);*/
-    }
-
     handleCheckboxChange(option: ItemType, $event)  {
-        console.log("handleCheckboxChange() called", option, $event);
-        //$event.target.checked;
-        this.updateCheckedValues();
+        let checked = $event.target ? $event.target.checked : false;
+        let index = this.checkedOptions.findIndex((checkedOption) => this.equalsCallback(checkedOption, option));
+        //console.log("Index: " + index);
+
+        //console.log("handleCheckboxChange() called", this.labelCallback(option), checked);
+        if (!checked && index != -1)
+            this.checkedOptions.splice(index, 1);
+        else
+            this.checkedOptions.push(option);
+        this.checkedOptionsChange.emit(this.checkedOptions);
     }
-
-    private updateCheckedValues(): void {
-        //this._zone.run(() => {
-            this.checkedOptions = this.getCheckedValues();
-            this.checkedOptionsChange.emit(this.checkedOptions);
-            //this.onCheckedChange.emit(this.checkedOptions);
-        //});
-        console.log("updateCheckedValues() done", this.checkedOptions);
-    }
-
-    private getCheckedValues(): string[] | ItemType[] {
-        let component = this;
-        let values = $(this.rootElement).find(":checkbox:checked").map(
-            function (i, domElement): string {
-                if (!domElement['checked'])
-                    return null;
-                let val = domElement["value"];
-
-                if (component.checkedOptionsAreValuesOnly)
-                    return val;
-
-                let clickedOption = component.valueToOptionMap.get(val);
-                if (!clickedOption)
-                    return console.warn("Unknown option clicked", val), null;
-                else
-                    return clickedOption;
```

Comment:
Well, manipulating the DOM like this is a good way to get NG to do random seeming things. :)